Validação XML Autorizacao xCampo

hbdbim opened the pull request:
Quando preenchido as lista obsCont e obsFisco da classe infAdic ao validar contra o schema estava retornando invalido.
Correção anotar xmlattribute nos campos xCampo:

```
    /// <summary>
    /// 
    /// </summary>
    public class obsFisco
    {
        /// <summary>
        ///     Z08 - Identificação do campo
        /// </summary>
        [XmlAttribute]
        public string xCampo { get; set; }

        /// <summary>
        ///     Z09 - Conteúdo do campo
        /// </summary>
        public string xTexto { get; set; }
    }
```

```
public class obsCont
    {
        /// <summary>
        ///     Z05 - Identificação do campo
        /// </summary>
        [XmlAttribute]
        public string xCampo { get; set; }

        /// <summary>
        ///     Z06 - Conteúdo do campo
        /// </summary>
        public string xTexto { get; set; }
    }
```

adeniltonbs committed:
Merge remote-tracking branch 'refs/remotes/origin/Danfe_NFe'

adeniltonbs committed:
Revert "Merge remote-tracking branch 'refs/remotes/origin/Danfe_NFe'"

This reverts commit c94c2ea2902af83efb8f7e8fe7b5f7ae20a2b729, reversing
changes made to 96732b780fdc00f09d51720872e220a72131312c.

adeniltonbs committed:
* Correção de erro ao deserializar um xml com a tag tpIntegra do grupo pag.card informada.

adeniltonbs committed:
Alterada versão do serviço RecepcaoEvento para NFCe do estado do PR, conforme issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/78

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Alterações segundo Nota Técnica 2015/002 Versão 1.20 de Novembro 2015;
* Arredondamento em alguns campos que ainda não estavam sendo tratados quanto ao limite de casas decimais.

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

adeniltonbs committed:
* Implementada Nota Técnica 2015/003, Versão 1.50.

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Adicionado método DanfeFrNfce.ExportarPdf(string arquivo) que possibilitará a exportação do DANFE da NFCe para PDF.

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Renomeado método ConfiguracaoApp.Salvar para ConfiguracaoApp.SalvarParaAqruivo para não causar confusão em relação ao propósito do método;
* Alterado texto do botão "Salvar Configurações" para "Salvar Configurações para arquivo".

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

adeniltonbs committed:
Atribuição de valores para alguns enuns das classes destTipos e ideTipos.

adeniltonbs committed:
Adicionado comentários na função ExportarPdf da classe NFe.Impressao.NFCe.FastReports.DanfeFrNfce

adeniltonbs committed:
* Corrreções na impressão do desconto do item.

adeniltonbs committed:
Implementado envio de nfe compactada para a versão 3.10.

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Adicionados endereços de consulta do QRCode para a SEFAZ de GO, conforme https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/108

adeniltonbs committed:
* Correção nos valores das informações de pagamento e cobrança da NFe/NFCe de demonstração;
* Descomentado grupo infNFeSupl na NFe/NFCe de demonstração.

rodrigomartins50 committed:
Merge pull request #1 from adeniltonbs/master

teste

rodrigomartins50 committed:
Implementado Administração CSC

adeniltonbs committed:
Merge pull request #118 from rodrigomartins50/master

Implementado Serviço Aministração CSC

adeniltonbs committed:
* Pequenas alterações na implementação do serviço de administração do CSC.

adeniltonbs committed:
+ Adicionado, na aplicação de demonstração, exemplo de consumo do serviço admCscNFCe;
* Alterado parâmetro "cnpj" do serviço admCscNFCe para "raizCNPJ", para que o consumo do serviço fique mais próximo do que é descrito no manual.

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Aumentado timeout padrão para 3000 milisegundos;
* Correção no método BtnAdminCsc_Click da classe MainWindow.

robertorp committed:
implementado exemplo de email

adeniltonbs committed:
* Correção do issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/125;
* Correção: a ordem de impresão dos produtos no DANFE não estava respeitando a ordem dos produtos no XML;
* Adicionado arredondamento no método get de todos os campos decimais. Isso fará com que a biblioteca corriga as decimais destes campos, quando for carregado um xml para classe.

adeniltonbs committed:
* Adicionado exemplo de leitura dos impostos de um XML que foi carregado para a classe.

adeniltonbs committed:
Merge pull request #119 from robertorp/master

Implementado envio de email

adeniltonbs committed:
* Adicionadas configurações para envio de email assíncrono e com o corpo em HTML;
* Adicionada documentação de código na classe EmailBuilder e ConfiguracaoEmail.

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Adicionado exemplo de envio de email com mensagem em HTML.

adeniltonbs committed:
* Correção: Definidos número de casas decimais para leitura de algumas propriedades da classe ICMSTot. Isso corrige o issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/148

adeniltonbs committed:
Alteração da versão do serviço RecepcaoEvento para a a SEFAZ do PR, conforme issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/85.

adeniltonbs committed:
* Correção conforme issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/135.

robertorp committed:
Adicionado no configuration manager para aceitar compilação x64 é x86
Adicionado Nuget

robertorp committed:
Deletado do configuration manager o x86 é x64
Alterado pastas do nuget

robertorp committed:
Ajustes no nuget
Adicionado nuget no readme

adeniltonbs commented:
Tranquilo. 
No TFS, clique em "Branches"
![image](https://cloud.githubusercontent.com/assets/6733483/16469266/7a16788c-3e26-11e6-948e-56a9b0086ad3.png) 
e selecione a branch master
![image](https://cloud.githubusercontent.com/assets/6733483/16469283/87620a38-3e26-11e6-9f6b-af8764cb19b5.png)

Se apresentar erros, baixe a biblioteca e adicione-a novamente ao TFS.

adeniltonbs committed:
* Removido Null propagator e nameof operator para manter compatibilidade com o c# 5.0

adeniltonbs committed:
Removido String Interpolation da classe NFe.Servicos.ServicosNFe, visando manter compatibilidade com o c# 5.0

Commit:
retirado projeto impressao do nuget

adeniltonbs committed:
Merge pull request #150 from robertorp/master

:+1:

adeniltonbs committed:
Adicionada descrição em alguns enuns;
Substituição da url de montagem do qrcode para a SEFAZ de GO, conforme https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/166

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

ernanisp commented:
Adailton da pra mandar a dll junto com o projeto em copy local coloca true, ele manda a dll necessaria para impressão. não ncessitando assim do usuario ter o fastreport

adeniltonbs commented:
Ok, vou fazer isso, mas antes vou separar a biblioteca de impressão em outro projeto,

adeniltonbs committed:
* Removida bilbioteca NFe.Impressao do projeto (por conta da dependência com o Fast Reports). Essa bilbioteca será liberada como um projeto à parte;
* Os métodos da classe Nfe.Impressao.EnderecadorDanfeNfce (ObterUrl e ObterUrlQrCode) foram movidos para classe de extensão NFe.Utils.InformacoesSuplementares.ExtinfNFeSupl;
* O enum TipoUrlDanfeNfce foi renomeado para TipoUrlConsultaPublica e foi movido para NFe.Classes.Enumns

adeniltonbs committed:
Merge

adeniltonbs committed:
Adicionado o projeto NFe.Integração à biblioteca. Colaboração de @AndrewsFB.
Obs: O projeto ainda tem algumas pendências para ficar 100% funcional.

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Implementado tratamento do parâmetro (Número do lote) do comando /enviar;
+ Adicionado parâmetro "tipo_documento" no comando /enviar, aceitando os seguintes valores: "D", "N" e "L", para indicar "Destinatário (nfeProc)", "Nfe" e "Lote (enviNFe3)";
* Correção no retorno do comando /recibo.

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Classe NFe.Utils.Auxiliar renomeada para Conversao;
+ Adicionada descrições nos itens do enum TipoEmissao;
+ Adicionada classe NFe.Utils.NfeSituacao com métodos para confirmar a situação da NFe, com base no cStat.

adeniltonbs committed:
Update README.md

Inclusão da descrição dos projetos NFe.Danfe.Base, NFe.Danfe.Fast e NFe.Danfe.AppTeste

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Pojetos de impressão do DANFE adiciondo novamente na solução Zeus NFe.

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

adeniltonbs committed:
- Removido diagrama NFe.Classes.cd

adeniltonbs committed:
* Correção de referência inválida ao tentar imprimir um DANFE em contingência.

adeniltonbs committed:
* Correção na versão do serviço RecepcaoEvento, conforme issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/187;
* Inclusão dos cStats 218 (NF-e já está cancelada na base de dados da SEFAZ) e 420 (Rejeição: Cancelamento para NF-e já cancelada) para o método NfeSituacao.Cancelada(int cStat).

ernanisp committed:
Versão inicial do serviço NFeDistribuicaoDFe

ernanisp committed:
Versão Inicial da manifestação do destinatario

ernanisp committed:
Melhorias nos servicços

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Merge pull request #197 from ernanisp/master

Criação dos Eventos Distribuição DFe e Manifestação destinatario

adeniltonbs committed:
* Enum ServicoNFe.RecepcaoEvento foi divido em quatro: RecepcaoEventoCancelmento, RecepcaoEventoCartaCorrecao, RecepcaoEventoEpec e RecepcaoEventoManifestacaoDestinatario. Essa divisão foi feita por conta que as SEFAZes mantêm diferentes urls e versões para esses serviços, emboram sejam todos do tipo RecepcaoEvento.

adeniltonbs committed:
* Adicionado enum para tipo de evento de manifestação do destinatário;
* Pequenos ajustes.

ernanisp committed:
Merge pull request #1 from adeniltonbs/master

Atualizado a partir da master original

adrbarros committed:
Ajusto para o Estado do Pará utilizar os serviços da SVRS (Sefaz Virtual Rio Grande do Sul) no envio e emissão na NFC-e

ernanisp committed:
Atualização das urls para qr-code de acordo com link (http://nfce.encat.org/desenvolvedor/qrcode/) divulgado na NT 2015/002

adeniltonbs committed:
Merge pull request #201 from ernanisp/master

Atualização urls QrCode

adeniltonbs committed:
Merge pull request #202 from adrbarros/master

Ajustes dos Serviços NFC-e Estado do Pará (PA)

adeniltonbs committed:
* Correção no comentário, somente :)

robertorp committed:
Retirando elemento xJust da carta correção

adeniltonbs committed:
* Desfeita alteração https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/commit/3548020024598418c83e519bf4e6e41bc87f3b65#diff-0fff1f2c6b3d7aa8a4febf7b9d533a5c.

adeniltonbs committed:
Merge pull request #206 from robertorp/master

Retirando elemento xJust da carta correção

marciojmuller committed:
Update refNFP.cs

- Campo CPF deve ser string visto que podem ter CPFs começando com digito 0
- Campo mod deve ser string visto que podem ter os valores 01 e  04
- Campo refCTe não pertence a tag refNFP (BA10) e sim a tag NFref (BA01)

adeniltonbs committed:
* Implementado tratamento do tamanho da impressão, via código, por conta de algumas impressoras térmicas que não possuem recurso para picotar ao final do documento. Ver issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/200#issuecomment-249669654;
* Substituído arquivo NFe.frx por arquivo enviado por fabinhoshz, praticamente pronto para impressão de NFe A4.

robertorp committed:
Implementado Envio de NVE

adeniltonbs committed:
Merge pull request #210 from robertorp/master

Implementado Envio de NVE

adeniltonbs committed:
Merge pull request #208 from marciojmuller/Update-no-arquivo-refNFP.cs

Update refNFP.cs

adeniltonbs committed:
* Adicionada configuração "ModoImpressao" que poderá ser definida como única página ou múltiplas páginas, visando atender um número maior de impressoras térmicas.

robertorp committed:
Atualizado arquivos do nuget, adicionado grupo do skype no readme.md

adeniltonbs committed:
Merge pull request #211 from robertorp/master

Atualizado arquivos do nuget, adicionado grupo do skype no readme.md

adeniltonbs committed:
* Usar CDATA para a url do qrCode, em vez de escape para &. Regra ZX02-22, NT2015.002, v141

adeniltonbs committed:
Merge branch 'master' of https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe

adeniltonbs committed:
* Método NFe.Servicos.ServicosNFe.TrocarAmpEscapePorCdata renomeado para NFe.Servicos.ServicosNFe.TrocarQrCodeAmpEscapePorCdata;
* Removido uso de null propragation, para manter compatibilidade com versões anteriores do C#.

adeniltonbs committed:
* Refatoração no commit 837dd5490731c44d0d2d84c5f7e75750d6612978;
* Correção de possível problema na classe CertificadoDigital.cs, reportado por Leandro Santos;
* Atualização dos schemas.

adeniltonbs committed:
* Responsabilidade pela serialização do qrCode no formato CDATA movida para classe infNFeSupl, pois toda vez que o qrCode for serializado(inclusive ao enviar para a SEFAZ), espera-se que esse atributo vá como CDATA.
Já o atributo qrCode ficará no formato original, pois dessa forma poderá ser usado para impressão sem necessidade de tratamento adicional.

adeniltonbs committed:
* Correção no método que salva as configurações do aplicativo de demonstração no disco.

adeniltonbs committed:
* Implementada interface IDisposable na classe ServicosNFe, para tentar resolver problemas com certificado A3 com a senha salva;
* Implementado consumo do Serviço NFeAutorizacao de forma síncrona (https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/134).

adeniltonbs committed:
Update README.md

adeniltonbs committed:
Update README.md

adeniltonbs committed:
* Correção do issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/217. Segundo a documentação, alguns caracteres especiais, como &, são aceitos em outras tags do xml, como na razão social da empresa. Por isso foi removida a substituição de escape desses caracteres, no método FuncoesXml.ClasseParaXmlString;
* Substituído método para serialização no formato CDATA do atributo qrCode, da classe infNFeSupl.

adeniltonbs committed:
* Adicionado valores para os enumeradores CSTCOFINS e CSTPIS, conforme issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/218;
* Implementada interface IXmlSerializable para a classe infNFeSupl. A abordagem anterior casusava problemas na impressão do DANFE da NFCe, no método do fastreports "RegisterData", que ficava em loop tentando varrer o grafo do elemento xml CDATA.

adeniltonbs committed:
* Correção conforme issue https://github.com/adeniltonbs/Zeus.Net.NFe.NFCe/issues/49#issuecomment-254216432.

adeniltonbs committed:
* Correção de problemas de conversão dps atributos numero e idlote

adeniltonbs committed:
* Removido acento do namespace NFe.Classes.Informacoes.Detalhe.Produto Especifico

adeniltonbs committed:
Removido null propagation de NFe.Danfe.Fast.NFCe.DanfeFrNfce, para manter compatibilidade com versões anterires do .NET Framework.

adeniltonbs committed:
* Renomeado namespace NFe.Classes.Informacoes.Detalhe.Produto Especifico para NFe.Classes.Informacoes.Detalhe.ProdEspecifico, para resolver problemas com espaços no VS2013
